### PR TITLE
fix(lemon-ui): Dashed `LemonDivider` shouldn't be black

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Raquel pointed out that `LemonDivider` is now black when dashed, which was not the case before. This is a regression from #14486, which changed the way the divider is colored. This regression should have been caught by UI snapshots, but we have a relatively large failure threshold set, which was supposed to avert some flakiness before.

## Changes

Restores the correct color of the divider. Also, we should no longer be having those reliability issues with snapshots, so reverting to the default failure threshold.

### CODE REVIEW

File: frontend/src/lib/lemon-ui/LemonDivider/LemonDivider.scss
Comment (raquelmsmith) [inline frontend/src/lib/lemon-ui/LemonDivider/LemonDivider.scss:1]: Don't forget the one below for the vertical border too!

Comment (Twixes) [reply frontend/src/lib/lemon-ui/LemonDivider/LemonDivider.scss:1]: Good point, done!

File: frontend/__snapshots__/scenes-app-insights--funnel-historical-trends--webkit.png
Comment (raquelmsmith) [inline frontend/__snapshots__/scenes-app-insights--funnel-historical-trends--webkit.png:1]: Not sure if this is an issue with snapshots or live on your branch but it looks like the carets in `Step 1` and `Step 3` changed. In fact, they look the same was as the "after" shot here on production... Which seems incorrect.

**Review by raquelmsmith — COMMENTED** [review]
A lot of the snapshots have no visible changes that I can see. I wonder if the threshold is now too low... If people start getting a lot of snapshot changes with no real changes they will likely ignore the snapshots completely, which then nullifies the point of having the snapshots. I wonder if something small like the border color accidentally changing is worth having a higher threshold so we can trust the snapshots more.

**Comment by Twixes** [comment]
Okay, clearly the scope exploded here. Back to basics: just divider colors. Looks like they did actually pop up in signup page snapshots, but we must have overlooked that.

File: frontend/__snapshots__/scenes-app-insights--trends-number.png
Comment (raquelmsmith) [inline frontend/__snapshots__/scenes-app-insights--trends-number.png:1]: Not sure why this one changed but I'm guessing just a snapshot fluke?

**Review by raquelmsmith — APPROVED** [review]
As long as the one unexpected snapshot change is just a fluke this LGTM

File: frontend/__snapshots__/scenes-app-insights--trends-number.png
Comment (Twixes) [reply frontend/__snapshots__/scenes-app-insights--trends-number.png:1]: Yeah, this is unfortunately indicative of an actual issue where _sometimes_ the number isn't sized properly, though this seems to be more prevalent in test. Re-running tests.

**Pull request merged**